File: GitHubPop/app/src/main/java/com/igormeira/githubpop/apirequests/GitHubService.java
package com.igormeira.githubpop.apirequests;

import com.igormeira.githubpop.model.PullRequest;
import com.igormeira.githubpop.model.Repositories;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GitHubService {

    // REPOSITORIES //
    @GET("search/repositories?q=language:Java&sort=stars")
    Call<Repositories> getRepositories(@Query("page") long page);

    // PULL REQUESTS //
    @GET("repos/{creator}/{name}/pulls")
    Call<List<PullRequest>> getPullRequests(@Path("creator") String repoCreator,
                                            @Path("name") String repoName);
}

File: GitRepositories/app/src/main/java/com/example/gitrepositories/model/services/GitHubService.kt
package com.example.gitrepositories.model.services

import com.example.gitrepositories.model.dto.PullRequest
import com.example.gitrepositories.model.dto.Repository
import com.example.gitrepositories.model.dto.RepositoryResponse
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GitHubService {

    @GET("search/repositories?q=language:Java&sort=stars")
    fun getRepositories(@Query("page") page: Int): Call<RepositoryResponse>

    @GET("repos/{creator}/{name}/pulls")
    fun getPullRequests(@Path("creator") repoCreator: String, @Path("name") repoName: String): Call<List<PullRequest>>

    companion object {
        fun getBaseService(): GitHubService {
            val retrofit: Retrofit = Retrofit.Builder()
                .baseUrl("https://api.github.com/")
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .build()
            return retrofit.create(GitHubService::class.java)
        }
    }
}